content/api/commands/origin.md

```diff
@@ -473,49 +473,114 @@ of
 into the callback.
 
-### Callback restrictions
+### Dependencies / Sharing Code
 
-Because of the way in which the callback is transmitted and executed, there are
-certain limitations on what code may be run inside it. In particular, the
-following Cypress commands will throw errors if used in the callback:
+It is not possible to use
+[CommonJS `require()`](https://nodejs.org/en/knowledge/getting-started/what-is-require/)
+or
+[dynamic ES module `import()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#dynamic_imports)
+within the callback. However, [`Cypress.require()`](/api/cypress-api/require)
+can be utilized to include [npm](https://www.npmjs.com/) packages and other
+files. It works exactly the same as
+[CommonJS `require()`](https://nodejs.org/en/knowledge/getting-started/what-is-require/).
 
-- `cy.origin()`
-- [`cy.intercept()`](/api/commands/intercept)
-- [`cy.session()`](/api/commands/session)
-- [`cy.server()`](/api/commands/server)
-- [`cy.route()`](/api/commands/route)
-- [`Cypress.Cookies.preserveOnce()`](/api/cypress-api/cookies)
+```js
+cy.origin('somesite.com', () => {
+  const _ = Cypress.require('lodash')
+  const utils = Cypress.require('../support/utils')
 
-It is also currently not possible to use
-[`require()`](https://nodejs.org/en/knowledge/getting-started/what-is-require/)
-or
-[dynamic `import()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#dynamic_imports)
-within the callback. Because of this limitation, it cannot use
-[npm](https://www.npmjs.com/) packages or other third-party libraries inside the
-callback, as there is no mechanism to reference them. This functionality will be
-provided in a future version of Cypress.
+  // ... use lodash and utils ...
+})
+```
+
+`Cypress.require()` can be used to share custom commands between tests run in
+primary and secondary origins. We recommend this pattern for setting up your
+[support file](/guides/core-concepts/writing-and-organizing-tests#Support-file)
+and setting up custom commands to run within the `cy.origin()` callback:
+
+`cypress/support/commands.js`:
+
+```js
+Cypress.Commands.add('clickLink', (label) => {
+  cy.get('a').contains(label).click()
+})
+```
+
+`cypress/support/e2e.js`:
+
+```js
+// makes custom commands available to all Cypress tests, outside of
+// cy.origin() callbacks
+import './commands'
+
+// code we only want run per test, so it shouldn't be run as part of
+// the execution of cy.origin() as well
+beforeEach(() => {
+  // ... code to run before each test ...
+})
+```
+
+`cypress/e2e/spec.cy.js`:
+
+```js
+it('tests somesite.com', () => {
+  cy.origin('somesite.com', () => {
+    // makes custom commands available to all subsequent
+    // cy.origin('somesite.com') calls
+    Cypress.require('../support/commands')
+
+    cy.visit('/page')
+    cy.clickLink('Click Me')
+  })
+})
+```
```

Comment:
Should we also add something around the same vein for just generic dependencies (figured I'd use `underscore` since Cypress ships with `lodash` and `jQuery`)?

```js
it('tests somesite.com', () => {
  cy.origin('somesite.com', () => {
    // makes the 'underscore' library available inside the closure
   const _ = Cypress.require('underscore')
   expect(_.isNumber(3)).to.be.true
  })
})
```


```js
it('tests somesite.com', () => {
  cy.origin('somesite.com', () => {
    // makes the 'underscore' library available inside the closure
   const _ = Cypress.require('underscore')
   expect(_.isNumber(3)).to.be.true
  })

  cy.origin('somesite.com', () => {
   // throws an error because `_` is undefined
   expect(_.isNumber(4)).to.be.true
  })
})
```

```js
it('tests somesite.com', () => {
  cy.origin('somesite.com', () => {
    // makes the 'underscore' library globally available inside the somesite.com origin context.
    _ = Cypress.require('underscore')
   expect(_.isNumber(3)).to.be.true
  })

  cy.origin('somesite.com', () => {
   // evaluates to true with no errors since `_` is bound to `window._`
   expect(_.isNumber(4)).to.be.true
  })
})
```

Reply:
I lean against it, since we'd essentially be demonstrating general JavaScript scoping semantics. Also, the global example could come off as an endorsement of that type of usage, which I don't think we necessarily want to promote.

Reply:
I think that is fair assessment. I still think it might be a good idea to include that `Cypress.require` works with general dependencies with a visual outside of implicitly endorsing possible bad practices, but I don't feel super strong about it either way.

Reply:
Looks like that is there too 😄